lib/embeddable/view_helpers.rb

```diff
@@ -4,29 +4,10 @@ def embed_video(embeddable, width, height, options = {})
       raise "Can't find embeddable name. Did you mean: \"#{embeddable.class.embeddables.last.inspect}\"?"
     end
     name = options[:name] || embeddable.class.embeddables.first
-    source = embeddable.send("#{name}_source")
-    render_embeddable_partial(embeddable, name, source, width, height)
+    render_embeddable_partial(embeddable, name, width, height)
   end
 
-  def render_embeddable_partial(embeddable, name, source, width, height)
-    attributes = { id: embeddable.send("#{name}_id"), width: width, height: height }
-    if embeddable.send("#{name}_on_youtube?")
-      render 'embeddable/partials/youtube', attributes
-    elsif embeddable.send("#{name}_on_vimeo?")
-      render 'embeddable/partials/vimeo', attributes
-    elsif embeddable.send("#{name}_on_dailymotion?")
-      render 'embeddable/partials/dailymotion', attributes
-    elsif embeddable.send("#{name}_on_veoh?")
-      render 'embeddable/partials/veoh', attributes
-    elsif embeddable.send("#{name}_on_vippy?")
-      render 'embeddable/partials/vippy', attributes
-    elsif embeddable.send("#{name}_on_liveleak?")
-      # not supported
-      link_to embeddable.send(source), embeddable.send(source)
-    elsif embeddable.send("#{name}_on_brightcove?")
-      render 'embeddable/partials/brightcove', attributes
-    else
-      link_to embeddable.send(source), embeddable.send(source)
-    end
+  def render_embeddable_partial(embeddable, name, width, height)
+    embeddable.send(name).render(self, width, height)
```

Comment:
Yes, that was cathartic. This is the least tested part of the code though, and is what I was referring to with the integration tests. I wanted to bring in ActionView so I could write a test that renders one of the services, but I couldn't get the Rails machinery working in a reasonable amount of time.